### PR TITLE
`General`: Sync progress bar color with theme

### LINKED ISSUE #5380: `Dark mode / Light Mode`: The color of the text in the participation bars and assessment bars in the course dashboard is not being updated automatically.
### Describe the bug

The color of the text in the participation bars and assessment bars in the course dashboard is not being updated automatically. It is necessary to approach the cursor. Check the video below. 

### To Reproduce

1. Go to Couse Management
2. Check that change between Dark/Light mode, the color of the text in the participation bars, and the assessment bars in the course dashboard is not being updated automatically. 

### Expected behavior

The color of the text in the participation bars and the assessment bars in the course dashboard should be updated automatically

### Screenshots

https://user-images.githubusercontent.com/88991954/178699154-081c0fdc-1313-4661-b967-47404bb10cb6.mov

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by bassner**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Fixes #5380.

### Description

- Subscribed to the theme service to automatically trigger updates
- Refactored the code so the template doesn't call functions anymore
- Added test for the component

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with a few exercises in different stages

Check that the progress bars still work and update with the theme automatically.

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| progress-bar.component.ts | 100% | 100% |

### Screenshots

ezgif-3-320252c4d9

### CODE REVIEW

**Review by luisa-we — APPROVED** [review]
Tested on TS1, works as expected!

**Review by alaraozdenler — APPROVED** [review]
Tested on ts2, works as expected

File: src/main/webapp/app/shared/dashboards/tutor-participation-graph/progress-bar/progress-bar.component.ts [hunk 1]
```diff
@@ -1,51 +1,73 @@
...
+    foregroundColorClass: string;
+    backgroundColorClass: string;
+    themeSubscription: Subscription;
+
+    constructor(private themeService: ThemeService, private ref: ChangeDetectorRef) {}
 
     ngOnInit() {
-        this.percentage = round(this.percentage);
+        this.themeSubscription = this.themeService.getCurrentThemeObservable().subscribe(() => {
+            this.chooseProgressBarTextColor();
```
Comment (ge65cer) [inline src/main/webapp/app/shared/dashboards/tutor-participation-graph/progress-bar/progress-bar.component.ts:24]: Do you need this method call here? `this.ref.detectChanges();` calls `ngOnChanges()` which in turn calls `chooseProgressBarTextColor()`, right?

**Review by ge65cer — APPROVED** [review]
Code looks good to me, beside one small comment.

File: src/main/webapp/app/shared/dashboards/tutor-participation-graph/progress-bar/progress-bar.component.ts (hunk 1 above)
Comment (bassner) [reply src/main/webapp/app/shared/dashboards/tutor-participation-graph/progress-bar/progress-bar.component.ts:24]: If we run a change detection cycle, but there are no changes to be detected, then ``ngOnChanges`` would not be called. So, in this case, we first need to apply the change ( = choose a new progress bar css class name and store it in the component field ), then check for changes to be applied.

Comment (bassner) [reply src/main/webapp/app/shared/dashboards/tutor-participation-graph/progress-bar/progress-bar.component.ts:24]: I tested it without that line and it does indeed not work :D

**Pull request merged**